docs/CLI_IMPLEMENTATION_PLAN.md

```diff
@@ -0,0 +1,627 @@
+# CLI Implementation Plan for Scuba API
+
+## Overview
+Create a command-line interface for the Scuba API that wraps the existing ScubaClient SDK functionality in a separate CLI folder.
+
+## Proposed CLI Structure
+
+### Main Commands
+```bash
+scuba [command] [subcommand] [options]
+
+# Metrics commands
+scuba metrics get --class <account|bucket|service|location> --resource <name> [--date YYYY-MM-DD]
+scuba metrics get-latest --class <account|bucket|service|location> --resource <name>
+scuba metrics batch --class <account|bucket|service|location> --resources <name1,name2> [--dates date1,date2]
+
+# Sub-metrics commands  
+scuba submetrics get --class <class> --resource <name> --type <type> --name <name> [--date YYYY-MM-DD]
+scuba submetrics get-latest --class <class> --resource <name> --type <type> --name <name>
```

Comment:
`--type` and `--name` not clear they apply to the sub metric here.

I'd rename `--subtype` `--subname`